pkg/cloud/services/networking/network.go

```diff
@@ -72,7 +72,10 @@ func (s *Service) ReconcileExternalNetwork(openStackCluster *infrav1.OpenStackCl
 
 	switch len(networkList) {
 	case 0:
-		return fmt.Errorf("external network not found")
+		// Not finding an external network is fine
```

Comment:
@hidekazuna

As part of this patch, I made a change to the external network reconciliation so that an installation can proceed even if an external network is not present.

I would prefer not to skip the external network reconciliation completely, as there is an edge case which it would be nice to support:

  * A shared "external" network is available that connects to the public internet
  * OpenStack provider manages an internal for the cluster and a router connecting it to the "external" network
  * Control-plane nodes also belong to a shared "control" network, which the management cluster is on

In this case, the router and external network are important for `LoadBalancer` services to work, but there is no need to expose the control-plane nodes to the internet as the management cluster can access them over the "control" network.

I've not tried to configure this with the existing options, and it may not be possible, but it would definitely be nice to support it in the future and skipping the external network reconciliation would mean it definitely won't work.

Reply:
Also, you might still want to have a bastion with a floating IP. This PR only prevents a floating IP being assigned to the API server.